File: android/src/main/kotlin/com/tpay/model/MerchantAuthorization.kt
package com.tpay.model

import org.json.JSONObject

data class MerchantAuthorization(
    val clientId: String,
    val clientSecret: String
) {
    companion object {
        private const val CLIENT_ID = "clientId"
        private const val CLIENT_SECRET = "clientSecret"

        fun fromJson(json: JSONObject): MerchantAuthorization {
            return MerchantAuthorization(
                clientId = json.getString(CLIENT_ID),
                clientSecret = json.getString(CLIENT_SECRET)
            )
        }
    }
}

File: lib/model/merchant/merchant_authorization.dart
import 'package:json_annotation/json_annotation.dart';

part 'merchant_authorization.g.dart';

/// Class responsible for storing merchant's [clientId] and [clientSecret].
@JsonSerializable()
class MerchantAuthorization {
  final String clientId;
  final String clientSecret;

  MerchantAuthorization({
    required this.clientId,
    required this.clientSecret
  });

  factory MerchantAuthorization.fromJson(Map<String, dynamic> json) => _$MerchantAuthorizationFromJson(json);
  Map<String, dynamic> toJson() => _$MerchantAuthorizationToJson(this);
}